### PR TITLE
Annotate `Socket` and related classes.

### PR DESCRIPTION
**Pull request by cpovirk**
Notes:

- `ServerSocket.setSocketFactory` "permits" `null` in the sense that
  passing `null` is a no-op if no factory has been set yet. If a factory
  has already been set, then passing `null` throws an exception. I
  figure that that means it's best annotated as non-nullable, since it's
  hard for me to imagine that anyone actually wants to pass `null`.
  (Possibly there was another method similar to this one, either in this
  PR or in another one? I forget.)
- The 2-arg `DatagramSocket` constructor says nothing about `null`. I
  did find [a test that checks that users can pass `null` successfully](https://github.com/openjdk/jdk/blob/9559e035d2692d9d61bec2a13b5239a98db077ac/test/jdk/java/net/DatagramSocket/Constructor.java#L75),
  though, and the handling of `null` seems in line with what we see in
  other `net` classes.

### CODE REVIEW

File: src/java.base/share/classes/java/net/MulticastSocket.java [hunk 1]
```diff
@@ -25,6 +25,10 @@
 
 package java.net;
 
+import org.jspecify.annotations.NullMarked;
```
Comment (wmdietl) [inline src/java.base/share/classes/java/net/MulticastSocket.java:28]: Did you mean to use this on the class?

**Review by wmdietl — COMMENTED** [review]
Looks good, just a few tweaks.

File: src/java.base/share/classes/java/net/MulticastSocket.java (hunk 1 above)
Comment (cpovirk) [reply src/java.base/share/classes/java/net/MulticastSocket.java:28]: Definitely, thanks. Do you think the rest of the annotations in the class are sufficient to just add it? (That had been my intention, and I'm pushing a commit to do so, but I'll hold off on committing until I hear back.)

Comment (wmdietl) [reply src/java.base/share/classes/java/net/MulticastSocket.java:28]: Yes, the annotations in this file look good to me - including that I don't know what the right answer for the TODOs is.

**Review by wmdietl — APPROVED** [review]

**Pull request merged**